crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -3946,26 +3946,33 @@ impl<'db> TypeInferenceBuilder<'db> {
             &alias.name.id
         };
 
+        // Avoid looking up attributes on a module if a module imports from itself
+        // (e.g. `from parent import submodule` inside the `parent` module).
+        let import_is_self_referential =
```

Comment:
This seems slightly more roundabout than it feels like it ought to have to be, considering that we must have already resolved `module_name` to a `File` in a method we called above to resolve `module_ty`. Or does `module_ty` itself know its `File`?

Reply:
jinx -- https://github.com/astral-sh/ruff/pull/18005#discussion_r2087012889

Reply:
`module_ty` should be a `ModuleLiteralType`, which wraps a `Module`, which knows its `File`. So can we compare that directly with `self.file()` instead of needing to do another `file_to_module` and compare module names?

Reply:
But if we are trying to merge this ASAP, we can also look into this as a follow-up.